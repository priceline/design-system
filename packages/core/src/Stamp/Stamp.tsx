import themeGet from '@styled-system/theme-get'
import React from 'react'
import styled, { css } from 'styled-components'
import {
  BorderRadiusProps,
  FontSizeProps,
  SpaceProps,
  borderRadius,
  compose,
  fontSize,
  space,
} from 'styled-system'
import type { ColorSchemeName } from '../theme'
import { borderRadiusAttrs } from '../utils/attrs'
import { applySizes, applyVariations, colorScheme, getPaletteColor, getTextColorOn } from '../utils/utils'

const sizes = {
  small: css`
    ${themeGet('textStyles.small')}
    line-height: ${themeGet('lineHeights.display')};
    min-height: 20px;
    & > svg {
      width: 12px;
      height: 12px;
    }
  `,
  medium: css`
    ${themeGet('textStyles.display0')}
    line-height: ${themeGet('lineHeights.standard')};
    min-height: 24px;
    & > svg {
      width: 14px;
      height: 14px;
    }
  `,
}

const variations = {
  outline: css`
    color: ${getPaletteColor('base')};
    border-color: ${(props) => getPaletteColor(props.borderColor, 'base')(props)};
    ${(props) => (props.bg ? `background-color: ${getPaletteColor(props.bg, 'base')(props)};` : '')}
  `,
  fill: css`
    color: ${getTextColorOn('base')};
    background-color: ${getPaletteColor('base')};
    border-color: ${getPaletteColor('base')};
  `,
  // todo: remove this copy of the fill variation in v4 as its name does not follow conventions
  solid: css`
    color: ${getTextColorOn('base')};
    background-color: ${getPaletteColor('base')};
    border-color: ${getPaletteColor('base')};
  `,
}

/**
 * @public
 */
export type StampProps = SpaceProps &
  FontSizeProps &
  BorderRadiusProps & {
    bg?: string
    borderColor?: string
    children?: React.ReactNode
    color?: string
    size?: 'small' | 'medium'
    variation?: 'outline' | 'fill' | 'solid'
    colorScheme?: ColorSchemeName
  }

/**
 * @public
 */
export const Stamp: React.FC<StampProps> = styled.div.attrs(borderRadiusAttrs)`
  display: inline-flex;
  align-items: center;
  vertical-align: top;
  min-height: 24px;
  text-transform: uppercase;
  letter-spacing: ${themeGet('letterSpacings.caps')};
  border-radius: 2px;
  border-width: 1px;
  border-style: solid;
  color: ${getPaletteColor('base')};

  ${({ theme }) => applySizes(sizes, undefined, theme.mediaQueries)};
  ${applyVariations('Stamp', variations)};
  ${colorScheme}

  ${(props) => compose(space, fontSize, borderRadius)(props)}
`

Stamp.displayName = 'Stamp'

Stamp.defaultProps = {
  px: 1,
  py: 0,
  color: 'border.light',
  bg: 'background.light',
  borderColor: 'border.base',
  borderRadius: 'md',
  size: 'medium',
  variation: 'outline',
}
