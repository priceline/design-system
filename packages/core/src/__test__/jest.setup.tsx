import '@testing-library/jest-dom'
import 'jest-styled-components'

global.requestAnimationFrame =
  global.requestAnimationFrame ||
  function _raf(cb) {
    return setTimeout(cb, 0)
  }
