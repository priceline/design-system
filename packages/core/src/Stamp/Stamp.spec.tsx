import { render, screen } from '../__test__/testing-library'
import { Stamp } from '../Stamp/Stamp'
import { theme } from '../theme/theme'

describe('Stamp', () => {
  let consoleError
  beforeEach(() => {
    consoleError = console.error
    console.error = jest.fn()
  })
  afterEach(() => (console.error = consoleError))

  test('renders', () => {
    const { asFragment } = render(<Stamp />)
    expect(asFragment()).toMatchSnapshot()
  })

  test('renders small size', () => {
    const { asFragment } = render(<Stamp size='small' />)
    expect(asFragment()).toMatchSnapshot()
  })

  test('renders medium fill', () => {
    const { asFragment } = render(<Stamp variation='fill' />)
    expect(asFragment()).toMatchSnapshot()
  })

  test('renders small fill', () => {
    const { asFragment } = render(<Stamp size='small' variation='fill' />)
    expect(asFragment()).toMatchSnapshot()
  })

  test('color gray sets text and icon color', () => {
    render(
      <Stamp color='gray' data-testid='stamp'>
        Stamp
      </Stamp>
    )
    const element = screen.getByTestId('stamp')
    expect(element).toHaveStyleRule('color', theme.colors.gray)
  })

  test('color blue sets text and icon color', () => {
    render(
      <Stamp color='blue' data-testid='stamp'>
        Stamp
      </Stamp>
    )
    const element = screen.getByTestId('stamp')
    expect(element).toHaveStyleRule('color', theme.colors.blue)
  })

  test('color green sets text and icon color', () => {
    render(
      <Stamp color='green' data-testid='stamp'>
        Stamp
      </Stamp>
    )
    const element = screen.getByTestId('stamp')
    expect(element).toHaveStyleRule('color', theme.colors.green)
  })

  test('color red sets text and icon color', () => {
    render(
      <Stamp color='red' data-testid='stamp'>
        Stamp
      </Stamp>
    )
    const element = screen.getByTestId('stamp')
    expect(element).toHaveStyleRule('color', theme.colors.red)
  })

  test('color orange sets text and icon color', () => {
    render(
      <Stamp color='orange' data-testid='stamp'>
        Stamp
      </Stamp>
    )
    const element = screen.getByTestId('stamp')
    expect(element).toHaveStyleRule('color', theme.colors.orange)
  })

  test('color purple sets text and icon color', () => {
    render(
      <Stamp color='purple' data-testid='stamp'>
        Stamp
      </Stamp>
    )
    const element = screen.getByTestId('stamp')
    expect(element).toHaveStyleRule('color', theme.colors.purple)
  })

  test('bg blue sets background color', () => {
    render(
      <Stamp bg='blue' data-testid='stamp'>
        Stamp
      </Stamp>
    )
    const element = screen.getByTestId('stamp')
    expect(element).toHaveStyleRule('background-color', theme.colors.blue)
  })

  test('borderColor green sets border color', () => {
    render(
      <Stamp borderColor='green' data-testid='stamp'>
        Stamp
      </Stamp>
    )
    const element = screen.getByTestId('stamp')
    expect(element).toHaveStyleRule('border-color', theme.colors.green)
  })

  test('has default borderGray border color', () => {
    render(<Stamp data-testid='stamp'>Stamp</Stamp>)
    const element = screen.getByTestId('stamp')
    expect(element).toHaveStyleRule('border-color', theme.colors.borderGray)
  })
})
