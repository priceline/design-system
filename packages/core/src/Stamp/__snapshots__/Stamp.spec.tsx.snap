// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stamp bg blue sets background color 1`] = `
.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  vertical-align: top;
  min-height: 24px;
  text-transform: uppercase;
  -webkit-letter-spacing: 0.025em;
  -moz-letter-spacing: 0.025em;
  -ms-letter-spacing: 0.025em;
  letter-spacing: 0.025em;
  border-radius: 2px;
  border-width: 1px;
  border-style: solid;
  color: #eceff2;
  font-size: 12px;
  font-weight: 700;
  line-height: 1.25;
  -webkit-letter-spacing: 0.025em;
  -moz-letter-spacing: 0.025em;
  -ms-letter-spacing: 0.025em;
  letter-spacing: 0.025em;
  text-transform: uppercase;
  line-height: 1.4;
  min-height: 24px;
  color: #eceff2;
  border-color: #c0cad5;
  background-color: #0068ef;
  padding-left: 4px;
  padding-right: 4px;
  padding-top: 0px;
  padding-bottom: 0px;
  border-radius: 8px;
}

.c0 > svg {
  width: 14px;
  height: 14px;
}

<div
  className="c0"
  color="border.light"
  size="medium"
/>
`;

exports[`Stamp borderColor green sets border color 1`] = `
.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  vertical-align: top;
  min-height: 24px;
  text-transform: uppercase;
  -webkit-letter-spacing: 0.025em;
  -moz-letter-spacing: 0.025em;
  -ms-letter-spacing: 0.025em;
  letter-spacing: 0.025em;
  border-radius: 2px;
  border-width: 1px;
  border-style: solid;
  color: #eceff2;
  font-size: 12px;
  font-weight: 700;
  line-height: 1.25;
  -webkit-letter-spacing: 0.025em;
  -moz-letter-spacing: 0.025em;
  -ms-letter-spacing: 0.025em;
  letter-spacing: 0.025em;
  text-transform: uppercase;
  line-height: 1.4;
  min-height: 24px;
  color: #eceff2;
  border-color: #0a0;
  background-color: #f4f6f8;
  padding-left: 4px;
  padding-right: 4px;
  padding-top: 0px;
  padding-bottom: 0px;
  border-radius: 8px;
}

.c0 > svg {
  width: 14px;
  height: 14px;
}

<div
  className="c0"
  color="border.light"
  size="medium"
/>
`;

exports[`Stamp color blue sets text and icon color 1`] = `
.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  vertical-align: top;
  min-height: 24px;
  text-transform: uppercase;
  -webkit-letter-spacing: 0.025em;
  -moz-letter-spacing: 0.025em;
  -ms-letter-spacing: 0.025em;
  letter-spacing: 0.025em;
  border-radius: 2px;
  border-width: 1px;
  border-style: solid;
  color: #0068ef;
  font-size: 12px;
  font-weight: 700;
  line-height: 1.25;
  -webkit-letter-spacing: 0.025em;
  -moz-letter-spacing: 0.025em;
  -ms-letter-spacing: 0.025em;
  letter-spacing: 0.025em;
  text-transform: uppercase;
  line-height: 1.4;
  min-height: 24px;
  color: #0068ef;
  border-color: #c0cad5;
  background-color: #f4f6f8;
  padding-left: 4px;
  padding-right: 4px;
  padding-top: 0px;
  padding-bottom: 0px;
  border-radius: 8px;
}

.c0 > svg {
  width: 14px;
  height: 14px;
}

<div
  className="c0"
  color="blue"
  size="medium"
/>
`;

exports[`Stamp color gray sets text and icon color 1`] = `
.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  vertical-align: top;
  min-height: 24px;
  text-transform: uppercase;
  -webkit-letter-spacing: 0.025em;
  -moz-letter-spacing: 0.025em;
  -ms-letter-spacing: 0.025em;
  letter-spacing: 0.025em;
  border-radius: 2px;
  border-width: 1px;
  border-style: solid;
  color: #4f6f8f;
  font-size: 12px;
  font-weight: 700;
  line-height: 1.25;
  -webkit-letter-spacing: 0.025em;
  -moz-letter-spacing: 0.025em;
  -ms-letter-spacing: 0.025em;
  letter-spacing: 0.025em;
  text-transform: uppercase;
  line-height: 1.4;
  min-height: 24px;
  color: #4f6f8f;
  border-color: #c0cad5;
  background-color: #f4f6f8;
  padding-left: 4px;
  padding-right: 4px;
  padding-top: 0px;
  padding-bottom: 0px;
  border-radius: 8px;
}

.c0 > svg {
  width: 14px;
  height: 14px;
}

<div
  className="c0"
  color="gray"
  size="medium"
/>
`;

exports[`Stamp color green sets text and icon color 1`] = `
.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  vertical-align: top;
  min-height: 24px;
  text-transform: uppercase;
  -webkit-letter-spacing: 0.025em;
  -moz-letter-spacing: 0.025em;
  -ms-letter-spacing: 0.025em;
  letter-spacing: 0.025em;
  border-radius: 2px;
  border-width: 1px;
  border-style: solid;
  color: #0a0;
  font-size: 12px;
  font-weight: 700;
  line-height: 1.25;
  -webkit-letter-spacing: 0.025em;
  -moz-letter-spacing: 0.025em;
  -ms-letter-spacing: 0.025em;
  letter-spacing: 0.025em;
  text-transform: uppercase;
  line-height: 1.4;
  min-height: 24px;
  color: #0a0;
  border-color: #c0cad5;
  background-color: #f4f6f8;
  padding-left: 4px;
  padding-right: 4px;
  padding-top: 0px;
  padding-bottom: 0px;
  border-radius: 8px;
}

.c0 > svg {
  width: 14px;
  height: 14px;
}

<div
  className="c0"
  color="green"
  size="medium"
/>
`;

exports[`Stamp color orange sets text and icon color 1`] = `
.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  vertical-align: top;
  min-height: 24px;
  text-transform: uppercase;
  -webkit-letter-spacing: 0.025em;
  -moz-letter-spacing: 0.025em;
  -ms-letter-spacing: 0.025em;
  letter-spacing: 0.025em;
  border-radius: 2px;
  border-width: 1px;
  border-style: solid;
  color: #f06f20;
  font-size: 12px;
  font-weight: 700;
  line-height: 1.25;
  -webkit-letter-spacing: 0.025em;
  -moz-letter-spacing: 0.025em;
  -ms-letter-spacing: 0.025em;
  letter-spacing: 0.025em;
  text-transform: uppercase;
  line-height: 1.4;
  min-height: 24px;
  color: #f06f20;
  border-color: #c0cad5;
  background-color: #f4f6f8;
  padding-left: 4px;
  padding-right: 4px;
  padding-top: 0px;
  padding-bottom: 0px;
  border-radius: 8px;
}

.c0 > svg {
  width: 14px;
  height: 14px;
}

<div
  className="c0"
  color="orange"
  size="medium"
/>
`;

exports[`Stamp color purple sets text and icon color 1`] = `
.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  vertical-align: top;
  min-height: 24px;
  text-transform: uppercase;
  -webkit-letter-spacing: 0.025em;
  -moz-letter-spacing: 0.025em;
  -ms-letter-spacing: 0.025em;
  letter-spacing: 0.025em;
  border-radius: 2px;
  border-width: 1px;
  border-style: solid;
  color: #70b;
  font-size: 12px;
  font-weight: 700;
  line-height: 1.25;
  -webkit-letter-spacing: 0.025em;
  -moz-letter-spacing: 0.025em;
  -ms-letter-spacing: 0.025em;
  letter-spacing: 0.025em;
  text-transform: uppercase;
  line-height: 1.4;
  min-height: 24px;
  color: #70b;
  border-color: #c0cad5;
  background-color: #f4f6f8;
  padding-left: 4px;
  padding-right: 4px;
  padding-top: 0px;
  padding-bottom: 0px;
  border-radius: 8px;
}

.c0 > svg {
  width: 14px;
  height: 14px;
}

<div
  className="c0"
  color="purple"
  size="medium"
/>
`;

exports[`Stamp color red sets text and icon color 1`] = `
.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  vertical-align: top;
  min-height: 24px;
  text-transform: uppercase;
  -webkit-letter-spacing: 0.025em;
  -moz-letter-spacing: 0.025em;
  -ms-letter-spacing: 0.025em;
  letter-spacing: 0.025em;
  border-radius: 2px;
  border-width: 1px;
  border-style: solid;
  color: #c00;
  font-size: 12px;
  font-weight: 700;
  line-height: 1.25;
  -webkit-letter-spacing: 0.025em;
  -moz-letter-spacing: 0.025em;
  -ms-letter-spacing: 0.025em;
  letter-spacing: 0.025em;
  text-transform: uppercase;
  line-height: 1.4;
  min-height: 24px;
  color: #c00;
  border-color: #c0cad5;
  background-color: #f4f6f8;
  padding-left: 4px;
  padding-right: 4px;
  padding-top: 0px;
  padding-bottom: 0px;
  border-radius: 8px;
}

.c0 > svg {
  width: 14px;
  height: 14px;
}

<div
  className="c0"
  color="red"
  size="medium"
/>
`;

exports[`Stamp has default borderGray border color 1`] = `
.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  vertical-align: top;
  min-height: 24px;
  text-transform: uppercase;
  -webkit-letter-spacing: 0.025em;
  -moz-letter-spacing: 0.025em;
  -ms-letter-spacing: 0.025em;
  letter-spacing: 0.025em;
  border-radius: 2px;
  border-width: 1px;
  border-style: solid;
  color: #eceff2;
  font-size: 12px;
  font-weight: 700;
  line-height: 1.25;
  -webkit-letter-spacing: 0.025em;
  -moz-letter-spacing: 0.025em;
  -ms-letter-spacing: 0.025em;
  letter-spacing: 0.025em;
  text-transform: uppercase;
  line-height: 1.4;
  min-height: 24px;
  color: #eceff2;
  border-color: #c0cad5;
  background-color: #f4f6f8;
  padding-left: 4px;
  padding-right: 4px;
  padding-top: 0px;
  padding-bottom: 0px;
  border-radius: 8px;
}

.c0 > svg {
  width: 14px;
  height: 14px;
}

<div
  className="c0"
  color="border.light"
  size="medium"
/>
`;

exports[`Stamp renders 1`] = `
.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  vertical-align: top;
  min-height: 24px;
  text-transform: uppercase;
  -webkit-letter-spacing: 0.025em;
  -moz-letter-spacing: 0.025em;
  -ms-letter-spacing: 0.025em;
  letter-spacing: 0.025em;
  border-radius: 2px;
  border-width: 1px;
  border-style: solid;
  color: #eceff2;
  font-size: 12px;
  font-weight: 700;
  line-height: 1.25;
  -webkit-letter-spacing: 0.025em;
  -moz-letter-spacing: 0.025em;
  -ms-letter-spacing: 0.025em;
  letter-spacing: 0.025em;
  text-transform: uppercase;
  line-height: 1.4;
  min-height: 24px;
  color: #eceff2;
  border-color: #c0cad5;
  background-color: #f4f6f8;
  padding-left: 4px;
  padding-right: 4px;
  padding-top: 0px;
  padding-bottom: 0px;
  border-radius: 8px;
}

.c0 > svg {
  width: 14px;
  height: 14px;
}

<div
  className="c0"
  color="border.light"
  size="medium"
/>
`;

exports[`Stamp renders medium fill 1`] = `
.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  vertical-align: top;
  min-height: 24px;
  text-transform: uppercase;
  -webkit-letter-spacing: 0.025em;
  -moz-letter-spacing: 0.025em;
  -ms-letter-spacing: 0.025em;
  letter-spacing: 0.025em;
  border-radius: 2px;
  border-width: 1px;
  border-style: solid;
  color: #eceff2;
  font-size: 12px;
  font-weight: 700;
  line-height: 1.25;
  -webkit-letter-spacing: 0.025em;
  -moz-letter-spacing: 0.025em;
  -ms-letter-spacing: 0.025em;
  letter-spacing: 0.025em;
  text-transform: uppercase;
  line-height: 1.4;
  min-height: 24px;
  color: #001833;
  background-color: #eceff2;
  border-color: #eceff2;
  padding-left: 4px;
  padding-right: 4px;
  padding-top: 0px;
  padding-bottom: 0px;
  border-radius: 8px;
}

.c0 > svg {
  width: 14px;
  height: 14px;
}

<div
  className="c0"
  color="border.light"
  size="medium"
/>
`;

exports[`Stamp renders small fill 1`] = `
.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  vertical-align: top;
  min-height: 24px;
  text-transform: uppercase;
  -webkit-letter-spacing: 0.025em;
  -moz-letter-spacing: 0.025em;
  -ms-letter-spacing: 0.025em;
  letter-spacing: 0.025em;
  border-radius: 2px;
  border-width: 1px;
  border-style: solid;
  color: #eceff2;
  font-size: 10px;
  font-weight: 700;
  line-height: 1.4;
  line-height: 1.25;
  min-height: 20px;
  color: #001833;
  background-color: #eceff2;
  border-color: #eceff2;
  padding-left: 4px;
  padding-right: 4px;
  padding-top: 0px;
  padding-bottom: 0px;
  border-radius: 8px;
}

.c0 > svg {
  width: 12px;
  height: 12px;
}

<div
  className="c0"
  color="border.light"
  size="small"
/>
`;

exports[`Stamp renders small size 1`] = `
.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  vertical-align: top;
  min-height: 24px;
  text-transform: uppercase;
  -webkit-letter-spacing: 0.025em;
  -moz-letter-spacing: 0.025em;
  -ms-letter-spacing: 0.025em;
  letter-spacing: 0.025em;
  border-radius: 2px;
  border-width: 1px;
  border-style: solid;
  color: #eceff2;
  font-size: 10px;
  font-weight: 700;
  line-height: 1.4;
  line-height: 1.25;
  min-height: 20px;
  color: #eceff2;
  border-color: #c0cad5;
  background-color: #f4f6f8;
  padding-left: 4px;
  padding-right: 4px;
  padding-top: 0px;
  padding-bottom: 0px;
  border-radius: 8px;
}

.c0 > svg {
  width: 12px;
  height: 12px;
}

<div
  className="c0"
  color="border.light"
  size="small"
/>
`;
